src/pyannote/metrics/diarization.py

```diff
@@ -646,3 +658,96 @@ def compute_components(
         return super(DiarizationCompleteness, self).compute_components(
             hypothesis, reference, uem=uem, **kwargs
         )
+
+
+class OverlappedDiarizationErrorRate(BaseMetric):
+    """Diarization error rate with details for overlap and non-overlap errors.
+    Error components will be prefixed with 'ovl' or 'nonovl' (e.g. 'ovl false alarm')
+
+    Parameters
+    ----------
+    collar : float, optional
+        Duration (in seconds) of collars removed from evaluation around
+        boundaries of reference segments.
+    """
+
+    OVDER_NAME = "diarization error rate"
+
+    def __init__(self, collar: float = 0.0):
+        super().__init__()
+
+        self.der_ovl = DiarizationErrorRate(collar=collar, skip_overlap=False)
+        self.der_nonovl = DiarizationErrorRate(collar=collar, skip_overlap=False)
+
+    @classmethod
+    def metric_components(cls) -> MetricComponents:
+        comps = []
+        for ovl in [OVLDER_PREFIX_NONOVL, OVLDER_PREFIX_OVL]:
+            for comp in [
+                MATCH_TOTAL,
+                MATCH_CORRECT,
+                MATCH_CONFUSION,
+                MATCH_MISSED_DETECTION,
+                MATCH_FALSE_ALARM,
+            ]:
+                comps.append(f"{ovl} {comp}")
+        return comps
+
+    @classmethod
+    def metric_name(cls) -> str:
+        return cls.OVDER_NAME
+
+    def compute_components(
+        self, reference: Annotation, hypothesis: Annotation, uem: Timeline | None = None
+    ) -> Details:
+
+        # map 'hypothesis' labels to 'reference' labels
+        mapping: dict = DiarizationErrorRate().optimal_mapping(
+            reference, hypothesis, uem=uem
+        )
+        hypothesis = hypothesis.rename_labels(mapping)
+
+        # split uem into non-overlapping and overlapping regions
+        overlap: Timeline = reference.get_overlap()
+        if uem is None:
+            uem = (
+                reference.support()
+                .get_timeline()
+                .union(hypothesis.support().get_timeline())
+            )
+        nonovl_regions: Timeline = uem.extrude(overlap)
+        ovl_regions: Timeline = uem.crop(overlap)
+
+        # update internal metrics for (non-)overlapping errors
+        comps_nonovl = self.der_nonovl.compute_components(
+            reference, hypothesis, uem=nonovl_regions
+        )
+        comps_ovl = self.der_ovl.compute_components(
+            reference, hypothesis, uem=ovl_regions
+        )
```

Comment:
Not sure I understand the logic here.

`compute_components` can actually lead to different mapping whether we focus on overlap or non overlap regions (because they have their own internal call to `optimal_mapping`). There is no garantee that this is the same mapping.

Or did I miss something?

Reply:
No, that's correct ... it was supposed to be an IdentificationErrorRate. But the tests passed because there was no overlap in the hypothesis. I think _now_ it should be correct.

Fixed it with the last two commits.